zerver/views/streams.py

```diff
@@ -318,13 +326,39 @@ def add_subscriptions_backend(
     if len(principals) > 0:
         if user_profile.realm.is_zephyr_mirror_realm and not all(stream.invite_only for stream in streams):
             return json_error(_("You can only invite other Zephyr mirroring users to invite-only streams."))
-        subscribers = set(principal_to_user_profile(user_profile, principal) for principal in principals)
+        subscribers = set(user_email_to_user_profile(user_profile, principal) for principal in principals)
     else:
         subscribers = set([user_profile])
 
+    for stream in created_streams:
```

Comment:
We only want to restrict newly created private stream without admins. As this function is also called to update existing stream subscription, we should only check this condition on created streams.